.github/workflows/phpunit.yml

```diff
@@ -0,0 +1,64 @@
+# https://help.github.com/en/categories/automating-your-workflow-with-github-actions
+
+name: "PHPUnit tests"
+
+on:
+    pull_request:
+    push:
+        branches:
+            - "master"
+
+jobs:
+    phpunit:
+        name: "PHPUnit tests"
+
+        runs-on: ${{ matrix.operating-system }}
+
+        strategy:
+            matrix:
+                dependencies:
+                    - "lowest"
+                    - "highest"
+                    - "locked"
+                php-version:
+                    - "7.2"
+                    - "7.3"
+                    - "7.4"
+                operating-system:
+                    - "ubuntu-latest"
+                    - "windows-latest"
+
+        steps:
+            - name: "Checkout"
+              uses: "actions/checkout@v2"
+
+            - name: "Install PHP"
+              uses: "shivammathur/setup-php@v2"
+              with:
+                  coverage: "pcov"
+                  php-version: "${{ matrix.php-version }}"
+                  ini-values: memory_limit=-1
+
+            - name: "Cache dependencies"
+              uses: "actions/cache@v2"
+              with:
+                  path: |
+                      ~/.composer/cache
+                      vendor
+                  key: "php-${{ matrix.php-version }}-${{ matrix.dependencies }}"
+                  restore-keys: "php-${{ matrix.php-version }}-${{ matrix.dependencies }}"
```

Comment:
`restore-keys` should have `php-${{ matrix.php-version }}-` instead, the original key is assumed.

Reply:
And the key is lacking `${{ hashFiles('composer.*') }}`

Reply:
I might open a PR with a comprehensive fix, if that's desirable.

Reply:
@sanmai if you think the caching is botched and you have the time/will to do so, please do :)

Reply:
Sure thing: #227